### PR TITLE
Separate mail templates from different mailers

### PR DESCRIPTION
**Pull request by raamSoftwire**
## Context

The "Candidate email send counts" export counts the number of emails sent, with one row per `mail_template`. However, some emails share the same `mail_template` name but are from different mailers, which means they have different content and different recipients. These emails are currently counted as one, but they need to be separated out.

## Changes proposed in this pull request

Update the export by changing the way the groupings work and adding a `mailer` column to make it explicit as to which mailer the `mail_template` is from.

## Guidance to review

Have I missed anything? Was this done on purpose?

## Link to Trello card

https://trello.com/c/66V90gyP/3245-investigate-why-email-notifications-data-extract-has-a-different-list-of-emails-sent-compared-to-qa

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/services/support_interface/candidate_email_send_counts_export.rb [hunk 1]
```diff
@@ -16,9 +17,9 @@ def data_for_export
     def emails_with_total_send_counts
       Email
         .select(
-          'mail_template, COUNT(mail_template) AS send_count, COUNT(DISTINCT "to") AS unique_recipients, MAX(created_at) AS last_sent_at',
+          'mail_template, COUNT(mail_template) AS send_count, COUNT(DISTINCT "to") AS unique_recipients, MAX(created_at) AS last_sent_at, mailer',
         )
-        .group('mail_template')
+        .group('mail_template, mailer')
```
Comment (raamSoftwire) [inline app/services/support_interface/candidate_email_send_counts_export.rb:22]: I used this post for reference here - https://stackoverflow.com/questions/1241352/rails-group-by-multiple-columns

File: app/exports/candidate_email_send_counts_export.yml [hunk 2]
```diff
@@ -17,3 +17,13 @@ custom_columns:
   unique_recipients:
     type: integer
     description: The number of unique recipients of this email
+
+  mailer:
+    type: string
+    description: The number of unique recipients of this email
```
Comment (duncanjbrown) [inline app/exports/candidate_email_send_counts_export.yml:23]:
```suggestion
    description: The Rails mailer which sent the email
```

**Review by duncanjbrown — APPROVED** [review]
This looks good — one documentation nit!

File: app/exports/candidate_email_send_counts_export.yml (hunk 2 above)
Comment (raamSoftwire) [reply app/exports/candidate_email_send_counts_export.yml]: Yep, I missed that. Cheers @duncanjbrown!

**Pull request merged**